flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -834,14 +817,82 @@ public void testUpdateUnknownInputChannel() throws Exception {
                             localResultPartitionId.getPartitionId(), localLocation));
 
             assertThat(
-                    inputGate.getInputChannels().get(remoteResultPartitionId.getPartitionId()),
+                    inputGate
+                            .getInputChannels()
+                            .get(createSubpartitionInfo(remoteResultPartitionId.getPartitionId())),
                     is(instanceOf((RemoteInputChannel.class))));
             assertThat(
-                    inputGate.getInputChannels().get(localResultPartitionId.getPartitionId()),
+                    inputGate
+                            .getInputChannels()
+                            .get(createSubpartitionInfo(localResultPartitionId.getPartitionId())),
                     is(instanceOf((LocalInputChannel.class))));
         }
     }
 
+    @Test
+    public void testUpdateUnknownChannelWithSubpartitionIndexRange()
```

Comment:
Maybe it's better to also cover the partition request with subpartition range scenario.

Reply:
Done